### PR TITLE
remove incomplete gradle wrapper infrastructure

### PR DESCRIPTION
**Pull request by kamilszymanski**
Since jar files (including gradle-wrapper.jar) have to be excluded from source releases current gradle wrapper infrastructure was incomplete rendering it unusable and confusing (I expected gradlew script to work without having to run default gradle task to download gradle-wrapper.jar).

### CODE REVIEW

**Comment by gwenshap** [comment]
this makes sense, but there are merge conflicts... can you rebase?

Also, can @ijuma or @granthenke confirm this makes sense?

File: .gitignore [hunk 1]
```diff
@@ -21,7 +21,6 @@ TAGS
 *.iml
 .project
 .settings
-.gradle
```
Comment (granthenke) [inline .gitignore:24]: Why should this be removed?

**Comment by granthenke** [comment]
@gwenshap This makes sense to me. We discussed it on @ijuma's recent Gradle 3.0 PR here: https://github.com/apache/kafka/pull/1774#issuecomment-241757166

I think it has lasted so long because people are unsure why it still exists in the first place.

File: .gitignore (hunk 1 above)
Comment (kamilszymanski) [reply .gitignore:24]: this entry is a duplicate of the one from line 5

**Comment by kamilszymanski** [comment]
@gwenshap rebased

**Comment by kamilszymanski** [comment]
Resolved conflicts after gradle update
CC @gwenshap @granthenke

**Comment by ijuma** [comment]
LGTM

**Pull request closed without merging**